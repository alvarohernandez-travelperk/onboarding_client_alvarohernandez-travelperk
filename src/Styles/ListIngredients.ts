import styled from 'styled-components'

export const Main = styled.div`
  padding: 20px 0px;
  border-top: 1px solid #a3a3a3;

  span {
    padding: 10px 0px;
  }
`;

export const InputBox = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const Button = styled.button`
    border: 1px solid #0e1012;
    border-radius: 5px;
    height: 40px;
    cursor: pointer;
    background-color: #0e1012;
    color: #eaeaea;
    margin: 0px 10px;
    &:hover:enabled{
      color: #0e1012;
      border-color: #0e1012;
      background-color: #eaeaea;
    }
`

export const List = styled.ul`
    list-style-type: none;
    padding: 20px;
    margin: 0px;

    li {
      padding: 2px;

      span {
        padding-left: 10px;
        cursor: pointer;
      }
    }
`