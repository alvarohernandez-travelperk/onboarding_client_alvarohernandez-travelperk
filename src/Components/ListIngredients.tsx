import { useState } from "react";
import InputField from "./InputField";
import { Button, Main, InputBox, List } from '../Styles/ListIngredients'
import { FaTrash } from 'react-icons/fa';


interface ListIngredientsProps {
  ingredients: string[];
  setIngredients: React.Dispatch<React.SetStateAction<string[]>>;
}

export default function ListIngredients({ ingredients, setIngredients }: ListIngredientsProps) {
  const [newIngredient, setNewIngredient] = useState<string>("")

  const handleNewIngredient = (e: React.ChangeEvent<HTMLInputElement>) => {
    setNewIngredient(e.target.value);
  };

  const handleIngredients = (newIngredient: string) => {
    const existIngredient = ingredients.find(el => el === newIngredient)
    if (!existIngredient) {
      const newArray = [...ingredients, newIngredient];
      setIngredients(newArray)
      setNewIngredient("");
    }
  };

  const deleteIngredient = (ingredientToDelete: string) => {
    const newArray = ingredients.filter(el => el !== ingredientToDelete);
    setIngredients(newArray)
  };

    return (
      <Main>
        <span>List of Ingredients :</span>
        <InputBox>
          <InputField
            handleInput={handleNewIngredient}
            value={newIngredient}
            placeholder="Type name here"
            title="Name"
          />
          <Button data-testid="add-ingredient-button" onClick={() => handleIngredients(newIngredient)}>Add Ingredient</Button>
        </InputBox>
        <List>
          {ingredients.map(ingredient =>
            <li data-testid="ingredient-item" key={ingredient}>
              {ingredient}
              <span onClick={() => deleteIngredient(ingredient)}>
              <FaTrash />
              </span>
            </li>
          )}
        </List>
      </Main>
    );
}