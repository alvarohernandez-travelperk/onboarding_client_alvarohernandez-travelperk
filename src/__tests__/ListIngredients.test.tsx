import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import ListIngredients from '../Components/ListIngredients';
import { BrowserRouter } from 'react-router-dom';

const setIngredients = jest.fn();
const mockIngredients = ["tomato", 'pasta', "carrots", "parmesan", "meat", "celery"]

afterEach(cleanup);

it("Snapshot Test ListIngredients Component", () => {
  const { asFragment } = render(<ListIngredients
    ingredients={mockIngredients}
    setIngredients={setIngredients} />,  {wrapper: BrowserRouter});
  expect(asFragment).toMatchSnapshot();
});

it("Test ListIngredients Texts", () => {
  const { getByTestId } = render(<ListIngredients
    ingredients={mockIngredients}
    setIngredients={setIngredients} />,  {wrapper: BrowserRouter});
    expect(getByTestId("add-ingredient-button")).toBeVisible();
    expect(screen.getAllByTestId("ingredient-item")).toHaveLength(mockIngredients.length);
});