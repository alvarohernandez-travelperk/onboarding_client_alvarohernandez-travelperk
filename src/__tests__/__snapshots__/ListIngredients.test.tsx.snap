// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test ListIngredients Component 1`] = `[Function]`;
